liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/InstallFeatureSupport.java

```diff
@@ -221,19 +222,26 @@ protected Set<String> getSpecifiedFeatures(String containerName) throws PluginEx
 
         if (util == null && noFeaturesSection) {
             //No features were installed because acceptLicense parameter was not configured
-            return new HashSet<String>();
+            return new FeaturesPlatforms();
         }
         else if (util == null && !noFeaturesSection) {
             Set<String> featuresToInstall = new HashSet<String>();
             for (Feature feature : features.getFeatures()) {
                 featuresToInstall.add(feature.toString());
             }
-            return featuresToInstall;
+            return new FeaturesPlatforms(featuresToInstall, new HashSet<String>());
         }
         else {
             Set<String> dependencyFeatures = getDependencyFeatures();
-            Set<String> serverFeatures = serverDirectory.exists() ? util.getServerFeatures(serverDirectory, getLibertyDirectoryPropertyFiles()) : null;
-            return util.combineToSet(pluginListedFeatures, dependencyFeatures, serverFeatures);
+            Set<String> serverFeatures = new HashSet<String>();
+            Set<String> serverPlatforms = new HashSet<String>();
+            FeaturesPlatforms getServerResult = serverDirectory.exists() ? util.getServerFeatures(serverDirectory, getLibertyDirectoryPropertyFiles()) : null;
+            if (getServerResult != null) {
+            	serverFeatures = getServerResult.getFeatures();
+            	serverPlatforms = getServerResult.getFeatures();
```

Comment:
I think you meant to call `getPlatforms()` when setting `serverPlatforms`.

Reply:
mistake - fixed